src/evaluations/analyzer.py

```diff
@@ -278,6 +282,60 @@ def __init__(self, out_dir, evaluation_directory, evaluation_run_name,
     super().__init__(out_dir, evaluation_directory, evaluation_run_name,
                      evaluation_name, estimable_criteria_list, plot_params)
 
+  def convert_raw_df_to_long_format(self):
+    """Convert the raw DataFrame to a long format.
+
+    Returns:
+      A pd.DataFrame. It is a long format of self.raw_df, which contains
+      columns of
+    """
+    # Get all the columns that are the true or estimated cardinality of all
+    # the frequency levels.
+    value_vars = []
+    for col in self.raw_df.columns:
+      if (simulator.TRUE_CARDINALITY_BASENAME in col
+          or simulator.ESTIMATED_CARDINALITY_BASENAME in col):
+        value_vars.append(col)
+
+    df_long = self.raw_df.melt(
+        id_vars=[SKETCH_ESTIMATOR_NAME, SCENARIO_NAME, simulator.RUN_INDEX,
+                 simulator.NUM_SETS],
+        value_vars=value_vars,
+        var_name=CARDINALITY_SOURCE,
+        value_name=CARDINALITY_VALUE,
+    )
+
+    def _split_source_and_frequency(source_freq):
```

Comment:
Would recommend making this module level. At the very least it should not be interspersed in the middle of a function.

Reply:
Done. Move it to be a (private) classmethod.